skypilot-org/skypilot
# [OCI] Support more OS types in addition to ubuntu

Besides ubuntu, there are many images in OCI, such as OracleLinux, RedHat, CentOS, etc. So supporting these images is a solid requirement. Esp. OracleLinux because in addition to ubuntu, there are GPU/HPC images are based on OracleLinux.   

This PR mainly well tested with OracleLinux.

This PR currently is OCI specific, will not affect other clouds. 

Tested (run the relevant ones):

- [v] Code formatting: `bash format.sh`
- [v] Any manual or new tests for this PR (please specify below)
   1. Add a OracleLinux image to the images.csv file, with OS field set to oraclelinux:
   2. Launch a cluster with more than one nodes. 
      For example, 
      Add a line in image.csv:
      skypilot:cpu-oraclelinux-9,us-sanjose-1,oraclelinux,9.4,ocid1.image.oc1.us-sanjose-1.aaaaaaaabuhcbcf627xexhgjlx26aq23fokzjws2l2dokxbt2f6jaspryk5q,,20240829
      sky launch -c "olcluster0" --num-nodes 2 --cloud oci --region us-sanjose-1 --cpus 2 --image-id skypilot:cpu-oraclelinux-9 --disk-size 256  "echo hi"
   4. Ensure the cluster can create successfully, and the task can be completed successfully.
   5. Check the system parameter are provisioned as expected, e.g. disk size. disk perf. 

   Test again with ubuntu image, to ensure the change does not affect the previous function with ubuntu image.

   Above tests are well passed.
      
- [ ] All smoke tests: `pytest tests/test_smoke.py` 
- [ ] Relevant individual smoke tests: `pytest tests/test_smoke.py::test_fill_in_the_name` 
- [ ] Backward compatibility tests: `conda deactivate; bash -i tests/backward_compatibility_tests.sh`
